services/agents/channel/channel_test.go

```diff
@@ -61,9 +62,13 @@ func setup(t *testing.T, connect func(*Channel) error, expected ...error) (agent
 		grpc.UnaryInterceptor(interceptors.Unary),
 		grpc.StreamInterceptor(interceptors.Stream),
 	)
+
+	channelM := &sync.Mutex{}
 	agentpb.RegisterAgentServer(server, &testServer{
 		connectFunc: func(stream agentpb.Agent_ConnectServer) error {
+			channelM.Lock()
```

Comment:
is there any race condition in this test?

Reply:
There was, I fixed it, thank you for your suggestion. :bow: